subgraphs/_reference_/src/sdk/protocols/perpfutures/protocol.ts

```diff
@@ -0,0 +1,581 @@
+   * @param volume {BigDecimal} The new total value locked for the protocol.
+   */
+  setVolume(volume: BigDecimal): void {
+    this.protocol.cumulativeVolumeUSD = volume;
+    this.save();
+  }
+
+  /**
+   * Adds a given USD value to the protocol's volume. It can be a positive or negative amount.
+   * Same as for setTotalValueLocked, this is mostly to be called internally by the library.
+   * But you can use it if you need to. It will also update the protocol's snapshots.
+   * @param volume {BigDecimal} The value to add to the protocol's volume.
+   */
+  addVolume(volume: BigDecimal): void {
+    this.protocol.totalValueLockedUSD =
+      this.protocol.totalValueLockedUSD.plus(volume);
+    this.save();
+  }
+
+  /**
+   * Sets the openInterest in USD for the protocol. Most times this will be called internally by
+   * other members of the library when openInterest changes are made to them. But if the library
+   * is not well fitted to a given protocol and you need to set the openInterest manually, you can
+   * use this method.
+   * It will also update the protocol's snapshots.
+   * @param openInterest {BigDecimal} The new total value locked for the protocol.
+   */
+  setOpenInterest(openInterest: BigDecimal): void {
+    this.protocol.openInterestUSD = openInterest;
+    this.save();
+  }
+
+  /**
+   * Adds a given USD value to the protocol InflowVolumeUSD. It can be a positive or negative amount.
+   * @param volume {BigDecimal} The value to add to the protocol's InflowVolumeUSD.
+   */
+  addInflowVolumeUSD(volume: BigDecimal): void {
+    this.protocol.cumulativeInflowVolumeUSD =
+      this.protocol.cumulativeInflowVolumeUSD.plus(volume);
+    this.save();
+  }
+
+  /**
+   * Adds a given USD value to the protocol ClosedInflowVolumeUSD. It can be a positive or negative amount.
+   * @param volume {BigDecimal} The value to add to the protocol's ClosedInflowVolumeUSD.
+   */
+  addClosedInflowVolumeUSD(volume: BigDecimal): void {
+    this.protocol.cumulativeClosedInflowVolumeUSD =
+      this.protocol.cumulativeClosedInflowVolumeUSD.plus(volume);
+    this.save();
+  }
+
+  /**
+   * Adds a given USD value to the protocol OutflowVolumeUSD. It can be a positive or negative amount.
+   * @param volume {BigDecimal} The value to add to the protocol's OutflowVolumeUSD.
+   */
+  addOutflowVolumeUSD(volume: BigDecimal): void {
+    this.protocol.cumulativeOutflowVolumeUSD =
+      this.protocol.cumulativeOutflowVolumeUSD.plus(volume);
+    this.save();
+  }
+
+  /**
+   * Adds a given USD value to the protocol supplySideRevenue. It can be a positive or negative amount.
+   * Same as for the rest of setters, this is mostly to be called internally by the library.
+   * But you can use it if you need to. It will also update the protocol's snapshots.
+   * @param rev {BigDecimal} The value to add to the protocol's supplySideRevenue.
+   */
+  addSupplySideRevenueUSD(rev: BigDecimal): void {
```

Comment:
similar thing I mentioned on the pools applies here

Reply:
I cannot make these functions private in the protocol because they are being used in other places.